api/v1alpha1/authorization_types.go

```diff
@@ -38,17 +42,35 @@ type AuthorizationRule struct {
 	// Action defines the action to be taken if the rule matches.
 	Action AuthorizationAction `json:"action"`
 
+	// Operation specifies the operation of a request, such as HTTP methods.
+	// If not specified, all operations are allowed or denied, based on the action of the rule.
+	//
+	// +optional
+	Operation *Operation `json:"operation,omitempty"`
+
 	// Principal specifies the client identity of a request.
 	// If there are multiple principal types, all principals must match for the rule to match.
 	// For example, if there are two principals: one for client IP and one for JWT claim,
 	// the rule will match only if both the client IP and the JWT claim match.
 	Principal Principal `json:"principal"`
 }
 
+// Operation specifies the operation of a request.
+// +notImplementedHide
+type Operation struct {
+	// Methods are the HTTP methods of the request.
+	// If multiple methods are specified, all specified methods are allowed or denied, based on the action of the rule.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	Methods []gwapiv1.HTTPMethod `json:"methods"`
```

Comment:
omitempty ? or we add that later ?

Reply:
It can't be empty becuase no other fields inside the Operation right now.  It should be optional/omitempty after paths or hosts are supported.